File: Data Structures & Algorithms/max-water-container/submission-1.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int i=0, j= heights.size()-1;
        int area = 0;

        while(i < j) {
            int greaterHeight = min(heights[i], heights[j]);
            area = max(area, greaterHeight * (j-i));
            if(heights[i] < heights[j]) i++;
            else j--;
        }
        return area;
    }
};

File: Data Structures & Algorithms/lowest-common-ancestor-in-binary-search-tree/submission-1.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution {
    public TreeNode LowestCommonAncestor(TreeNode root, TreeNode p, TreeNode q) {
        if(root is null || p is null || q is null) {
            return null;
        }
        if(Math.Max(p.val, q.val) < root.val) {
            return LowestCommonAncestor(root.left, p, q);
        }
        else if(Math.Min(p.val, q.val) > root.val) {
            return LowestCommonAncestor(root.right, p, q);
        }
        else {
            return root;
        }
    }
}
